### ISSUE TITLE
'braid push' leads to 'patch failed' on Windows

### ISSUE THREAD

**Issue by 0x084E** [cristibalan/braid#59]
I've got the mentioned error message  while trying to push changes to a repo .I looked a bit into the code and first removed the error message splitting in ```operations.rb:19``` to get a full message. This outcropped the additional messge ```patch does not apply``` which led me to the stackoverflow thread http://stackoverflow.com/questions/4770177/git-patch-does-not-apply where somebody suggests o call ```git apply``` with the parameters ```--ignore-space-change --ignore-whitespace```.
So I replaced ```--whitespace=nowarn``` with these parameters and was able to push without problems. I also tried the mentioned ```--whitespace=fix``` but this seems to replace all CRLF with LF.

Long story short, I would appreciate if the whitespace problem could be fixed in any way. Always getting the full error messages and not only the first line would be nice, too.

P.S.:
Thanks for a nice tool! Braid is the best and easiest submodule alternative i've seen so far.

**Comment by 0x084E** [comment]
As it turned out using ```--ignore-space-change --ignore-whitespace``` seems to mash up line endings, too.
Could it be an issue with having ```autocrlf = false```?

**Comment by 0x084E** [comment]
After further tests it seems to me that this was caused by some mixed mode line ending files. I'm still have dubts about what the problem excatly but it seems to be repository related.
I'm sorry for the confusion...

**Comment by 0x084E** [comment]
I think I've tracked it down to be caused by git generating patches with ln line endings always.
Any hint on how to solve this without setting autocrlf to true?

**Comment by realityforge** [comment]
Unfortunately I have no real solution for this.

We have been forced recently make all of our developers use `autocrlf = false`, explicitly set EOL style in `.gitattributes` and then explicitly check EOLs as a build step :( It sucks but I am not sure what the solution is.

**Comment by realityforge** [comment]
Next time I work on Braid I will try to display the full error messages

**Comment by 0x084E** [comment]
> We have been forced recently make all of our developers use autocrlf = false, explicitly set EOL style in .gitattributes and then explicitly check EOLs as a build step :( It sucks but I am not sure what the solution is.

After reading a lot about git and line endings I came to the same conclusion (:. Anyway, thanks for your confirmation. I'll go with a ```.gitattributes``` file.

> Next time I work on Braid I will try to display the full error messages

Nice, thanks!

**Comment by mattmccutchen** [comment]
This sounds like the kind of problem I was hoping to avoid by changing `braid push` to use a direct three-way tree merge instead of going through a patch file.

**Comment by mattmccutchen** [comment]
@0x084E: I'd be curious if #60 fixes the problem you saw.

**Comment by 0x084E** [comment]
@mattmccutchen I've tried your pull request but I can't get it to work. I'm getting the following errors:

```
Braid: Shell error: error: object directory C:/tmp/braid-test/Repo1/.git/objects? does not exist; check .git/objects/info/alternates.
error: object directory C:/Users/User/.braid/cache/C__tmp_braid-test_Repo2/objects? does not exist; check .git/objects/info/alternates.
fatal: failed to unpack tree object ceb1001f34e43c649c47ce3ef136d0745be0259e
```

**Comment by mattmccutchen** [comment]
@0x084E: Oops.  I guess I need to actually set up a development environment on Windows so I can try this for myself.  Looks like the problem is just that the alternates file needs to use LF, not CRLF, even on Windows.  How about #61?

**Comment by 0x084E** [comment]
Confirmed, in version 1.0.20 the problem is solved. Thanks to everyone involved!
